File: kits/python/agent.py
import numpy as np
import sys


# -----------------------------------
# USEFUL CONSTANTS

TEMP = -1
UNOCCUPIED = 0
TAIL = 1
TERRITORY = 2
BOMB = 3
BOOST = 4

# -----------------------------------

class Agent():
    def __init__(self, player: str) -> None:
        self.player = player
        self.times_moved = 0

    def formatAction(self, turn: int):
        return { 'turn': turn }

    def act(self, iter: int, curr_step: int, obs, remainingOverageTime: int = 60):

        # get all info; note: _dones and _infos are irrelevant here
        obs, rewards, _dones, _infos = obs

        # if not allowed to move, don't waste time computing
        if (obs[self.player]['resetting']):
            return self.formatAction(0)

        # if first iteration, save constant observations like player_num
        if (iter == 0 and curr_step == 0):
            self.num = obs[self.player]['player_num']

        # separate player observation from board observation
        me_obs = obs[self.player]
        board_obs = obs['board']

        # collection essential player info
        direction = me_obs['direction']
        head = me_obs['head']
        energy = me_obs['energy']
        speed = me_obs['speed']

        # save arrays which describe board state to numpy arrays for processing
        board = np.array(board_obs['board_state'])
        player_owned = np.array(board_obs['players_state'])


        # simple action: turn every 10 iterations
        turn = 1 if self.times_moved % 10 == 0 and iter != 0 else 0
        
        # increment steps
        self.times_moved += 1

        # note action should be a dict with action['turn'] = -1, 0, or 1
        return self.formatAction(turn)

File: kits/java/src/main/java/com/elementai/Agent.java
package com.elementai;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.elementai.element.Player;

import java.util.Map;
// import com.elementai.objectmapper.Mapper;

// import java.util.Random;

public class Agent {    
    public final int TEMP = -1;
    public final int UNOCCUPIED = 0;
    public final int TAIL = 1;
    public final int TERRITORY = 2;
    public final int BOMB = 3;
    public final int BOOST = 4;


    public int iter;
    public int currStep;
    public int remainingOverageTime;
    public String player;

    public Map<String, Player> obs;
    public int[][] boardState;
    public int[][] playersState;

    public Map<String, Integer> rewards;

    private int timesMoved = 0;

    public String act() throws JsonProcessingException {

        /* -----------------------------------------------------
         * DO NOT CHANGE ANY CODE ABOVE THIS LINE
         */ 


        /*
         * YOUR BOT GOES HERE. Remember to set turn = -1, 0, or 1
         */

        Player p = obs.get(player);

        // if we are not allowed to go, don't waste time computing
        if (p.speed <= currStep || p.resetting)
            return formatAction(0);

        // bot makes a simple square and stays in it
        int turn = timesMoved != 0 ? timesMoved % 10 == 0 ? 1 : 0 : 0;

        timesMoved++;

        // NOTE: ALWAYS use formatAction() when returning a turn value
        return formatAction(turn);
    }

    public String formatAction(int turn) {
        return "{\"turn\": " + turn + "}";
    }

}